src/common/VirtualTerminal.cpp

```diff
@@ -38,14 +42,47 @@
 
 namespace SDDM {
     namespace VirtualTerminal {
+#ifdef __FreeBSD__
+        static const char *defaultVtPath = "/dev/ttyv0";
+
+        QString getVtPath(int vt) {
+            char c = (vt <= 10 ? '0' : 'a') + (vt - 1);
+            return QStringLiteral("/dev/ttyv%1").arg(c);
+        }
+
+        int getVtActive(int fd) {
+            int vtActive = 0;
+            if (ioctl(fd, VT_GETACTIVE, &vtActive) < 0) {
+                qCritical() << "Failed to get current VT:" << strerror(errno);
+                return -1;
+            }
+            return vtActive;
+        }
+#else
+        static const char *defaultVtPath = "/dev/tty0";
```

Comment:
tty0 is not a default vt path, it's the active vt. I'd suggest calling it s_activeVt.

Reply:
It's not the active VT on FreeBSD; ttyv0 is VT 1, ttyv1 is VT 2 etc on FreeBSD. There is no equivalent of Linux's tty0 on FreeBSD. Hence why I called it "default" as some kind of "if you don't need a specific one and just want to do queries then use this one" device.

Reply:
So, which version should I choose to get this merged?

Reply:
Let's leave s_defaultVt then :facepalm: